walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,129 @@
+  }
+
+
+  /**
+   * Checks if the time between the event and the last event is enough for the requested meeting
+   * time
+   *
+   * @param availableTimes A collection of times which will work for the request to be added to
+   * @param event The  event which we will check to see if there is enough time between its start
+   *     and the last meeting's end
+   * @param request The specifications for the requested meeting
+   * @param optional A boolean which determines if we are checking optional attendees as well
+   * @param start A number which represents the end of the last meeting
+   *
+   * @return The end of this event if applicable
+   */
+  public static int checkIfEventConflicts(Collection<TimeRange> availableTimes, Event event,
+      MeetingRequest request, boolean optional, int end) {
+    if (hasAttendees(event.getAttendees(), request, optional)) {
+      TimeRange eventRange = event.getWhen();
+
+      //First check that the end of the last event is before the start of the next
+      if (end < eventRange.start()) {
+        TimeRange range = TimeRange.fromStartEnd(end, eventRange.start(), false);
+
+        //Check if the time between these two events has a long enough duration
+        if (range.duration() >= request.getDuration()) {
+          availableTimes.add(range);
+        }
+      }
+
+      //Return the end of this event if it ends after the event before
+      if (end < eventRange.end()) {
+        return eventRange.end();
+      }
+    }
+
+    //Else return the end of the prior event
+    return end;
+  }
+
+  /**
+   * Checks if the set of event attendees has any members from the request
+   *
+   * @param eventAttendees The set of people attending the event
+   * @param request The specifications of the request used to retrieve attendees and optional
+   *     attendees
+   * @param checkingOptional A boolean if we are checking for optional attendees
+   * @return If any of the requested attendees exist in the list for eventAttendees
+   */
+  public static boolean hasAttendees(
+    Set eventAttendees, MeetingRequest request, boolean checkingOptional) {
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    if (checkingOptional) {
+        return !Collections.disjoint(attendees, eventAttendees) || !Collections.disjoint(optionalAttendees, eventAttendees);
+    } else {
+        return !Collections.disjoint(attendees, eventAttendees);
+    }
+  }
+
+  /**
+   * Adds the time to the collection if the time between the final meeting and the end of the
+   * day if there is enough time
+   * 
+   * @param timeOfLastMeeting The end time of the last meeting
+   * @param request The requested meeting specifications
+   * @param availableTimes The collection of times to add to
+   */
+  public static void addFinalRange(int timeOfLastMeeting, MeetingRequest request, Collection<TimeRange> availableTimes) {
```

Comment:
Can this method be private?  Or is it called from some other class?

Reply:
privatized